src/apps/aurora-plus/aurora/aurora-plus.stake.contract-position-fetcher.ts

```diff
@@ -0,0 +1,63 @@
+import { Inject } from '@nestjs/common';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { getImagesFromToken } from '~app-toolkit/helpers/presentation/image.present';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { ContractPosition } from '~position/position.interface';
+import { ContractType } from '~position/contract.interface';
+import { claimable, supplied } from '~position/position.utils';
+import { Network } from '~types/network.interface';
+
+import { AURORA_PLUS_DEFINITION } from '../aurora-plus.definition';
+import { AuroraPlusContractFactory } from '../contracts';
+import { range } from 'lodash';
+
+const appId = AURORA_PLUS_DEFINITION.id;
+const groupId = AURORA_PLUS_DEFINITION.groups.stake.id;
+const network = Network.AURORA_MAINNET;
+
+const AURORA_ADDRESS = "0x8bec47865ade3b172a928df8f990bc7f2a3b9f79".toLowerCase();
+const AURORA_STAKING_ADDRESS = "0xccc2b1ad21666a5847a804a73a41f904c4a4a0ec".toLowerCase();
+
+@Register.ContractPositionFetcher({ appId, groupId, network })
+export class AuroraAuroraPlusStakeContractPositionFetcher implements PositionFetcher<ContractPosition>
+{
+  constructor(
+    @Inject(APP_TOOLKIT)
+    private readonly appToolkit: IAppToolkit,
+    @Inject(AuroraPlusContractFactory)
+    private readonly auroraPlusContractFactory: AuroraPlusContractFactory,
+  ) { }
+
+  async getPositions() {
+    const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+    let tokens:any = [];
+    const aurora = baseTokens.find(t => t.address === AURORA_ADDRESS)!;
+    if (!aurora) tokens.push(supplied(aurora));
```

Comment:
The logic is broken here... you'll never push the Aurora token to your underlying tokens array because fo this line. Change this to:

```ts
if (!aurora) return []; // handle the case if we don't support Aurora (this should never happen since we do support it)
tokens.push(aurora)
```

Reply:
that was meant to be !! . fixed